baseapp/msg_service_router.go

```diff
@@ -50,20 +55,42 @@ func (msr *MsgServiceRouter) RegisterService(sd *grpc.ServiceDesc, handler inter
 		fqMethod := fmt.Sprintf("/%s/%s", sd.ServiceName, method.MethodName)
 		methodHandler := method.Handler
 
+		var requestTypeName string
+
+		// NOTE: This is how we pull the concrete request type for each handler for registering in the InterfaceRegistry.
+		// This approach is maybe a bit hacky, but less hacky than reflecting on the handler object itself.
+		// We use a no-op interceptor to avoid actually calling into the handler itself.
+		_, _ = methodHandler(nil, context.Background(), func(i interface{}) error {
+			msg, ok := i.(proto.Message)
+			if !ok {
+				// We panic here because there is no other alternative and the app cannot be initialized correctly
+				// this should only happen if there is a problem with code generation in which case the app won't
+				// work correctly anyway.
+				panic(fmt.Errorf("can't register request type %T for service method %s", i, fqMethod))
+			}
+
+			msgName := proto.MessageName(msg)
+			requestTypeName = fmt.Sprintf("/%s", msgName)
```

Comment:
@aaronc What was the reason you chose to add a "/" prefix? afaict we can just use proto.MessageName as key.

Reply:
Just to note that the `/` prefix causes some confusion for me:
- baseapp's msg_service_router indexes routes with a `/`
- however querying events `message.action='%s'` does not expect a `/`
- in authz/feegrant right now I put a `/` prefix everywhere (incl in CLI, state etc...) to match the msg_service_router

TBH it would be ideal to decide on one convention, and use it everywhere. I would propose to remove the `/`.

Reply:
Using a leading `/` (aka sdk.MsgTypeURL) everywhere. See https://github.com/cosmos/cosmos-sdk/pull/9139#issuecomment-827503574.